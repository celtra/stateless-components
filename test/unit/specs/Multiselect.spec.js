import Multiselect from '@/stateless/multiselect.vue'
import Vue from 'vue'
import snapshotMixin from '../component_snapshot_mixin'

const Constructor = Vue.extend(Multiselect)
let vm = null

Vue.component('transition-group', {
    render (createElement) {
        return createElement('div', this.$slots.default)
    },
})

describe('Multiselect', () => {
    beforeEach(function () {
        vm = new Constructor({
            propsData: {
                value: [],
                options: [
                    { id: 1, label: "option 1" },
                    { id: 2, label: "option 2" },
                    { id: 3, label: "option 3" },
                ],
                isSearchable: true,
            },
        }).$mount()
    })

    snapshotMixin(Multiselect)

    describe('computed', () => {
        describe('allOptions', ()=>{
            it('should filter options', function () {
                expect(vm.allOptions.length).toBe(3)
                vm.searchQuery = "2"
                expect(vm.allOptions.length).toBe(1)
                expect(vm.allOptions[0].label).toBe("option 2")

                vm.searchQuery = "option"
                expect(vm.allOptions.length).toBe(3)

                vm.searchQuery = "option "
                expect(vm.allOptions.length).toBe(3)
            })
        })
    })

    describe('methods', () => {
        describe('isChecked', () => {
            it('should be checked if checked contains id', function () {
                expect(vm.isChecked(1)).toBe(false)
                vm.value.push(1)
                expect(vm.isChecked({ id: 1 })).toBe(true)
            })
        })

        describe('setChecked', () => {
            it('should set checked', function () {
                spyOn(vm, '$emit')
                expect(vm.isChecked({ id: 1 })).toBe(false)
                vm.setChecked({ id: 1 }, true)
                expect(vm.$emit).toHaveBeenCalledWith('input', [1])
            })
        })

        describe('setMultiple', () => {
            it('should select all', function () {
                spyOn(vm, '$emit')

                expect(vm.isChecked({ id: 1 })).toBe(false)
                expect(vm.isChecked({ id: 2 })).toBe(false)
                expect(vm.isChecked({ id: 3 })).toBe(false)
                vm.setMultiple(true)
                expect(vm.$emit).toHaveBeenCalledWith('input', [1, 2, 3])
            })

            it('should clear all', function () {
                spyOn(vm, '$emit')

                vm.value = ['1', '2', '3']
                vm.setMultiple(false)
                expect(vm.$emit).toHaveBeenCalledWith('input', [])
            })
        })
    })
})
