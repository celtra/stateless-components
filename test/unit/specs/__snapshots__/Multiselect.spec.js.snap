// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiselect should match the snapshot 1`] = `
<div
  class="multiselect"
>
  <div
    class="multiselect__search-with-icon"
  >
    <div
      class="input input--phat input--dark"
      id="search"
    >
      <div
        class="input__icon-prepend"
      >
        <img
          class="input__icon"
          src="/img/icons/search.svg"
        />
      </div>
       
      <div
        class="input-field input-field--with-icon"
        title=""
      >
        <div
          class="input-field__overlay"
        >
          
            Search
        
        </div>
         
        <div
          class="input-field__label-text"
        >
          
            
        
        </div>
         
        <div
          class="input-row"
        >
          <!---->
           
          <div
            class="input-row__input-flex"
          >
            <!---->
             
            <!---->
             
            <input
              class="input-row__placeholder-text"
              placeholder="Search"
              type="text"
            />
          </div>
           
          <!---->
           
          <!---->
        </div>
         
        <div
          class="input-field__border-overlay"
        />
         
        <div
          class="input-field__message-wrap"
        >
          <div
            class="input-field__helper-text"
          >
            
                    
                
          </div>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="multiselect__change-multiple"
  >
    <div
      class="checkbox-element multiselect__select-all checkbox-element--condensed checkbox-element--dark"
      data-id="select-all"
      tabindex="0"
      title="SELECT ALL"
    >
      <div
        class="checkbox-element__check-row"
        title="SELECT ALL"
      >
        <div
          class="checkbox-element__check-wrapper"
        >
          <div
            class="checkbox-element__square"
          />
           
          <div
            class="checkbox-element__check"
          />
        </div>
         
        <div
          class="checkbox-element__label-text"
        >
          <span
            class="multiselect__select-all-label"
          >
            SELECT ALL
          </span>
        </div>
      </div>
       
      <!---->
    </div>
  </div>
   
  <div
    class="multiselect__options"
    style="height: 0px;"
  >
    <div
      name="multiselect__item"
      tag="div"
    >
      <div
        class="multiselect__option"
      >
        <div
          class="checkbox-element multiselect__checkbox checkbox-element--normal checkbox-element--dark"
          data-id="option-1"
          tabindex="0"
          title="option 1"
        >
          <div
            class="checkbox-element__check-row"
            title="option 1"
          >
            <div
              class="checkbox-element__check-wrapper"
            >
              <div
                class="checkbox-element__square"
              />
               
              <div
                class="checkbox-element__check"
              />
            </div>
             
            <div
              class="checkbox-element__label-text"
            >
              
                    option 1
                
            </div>
          </div>
           
          <!---->
        </div>
         
        <!---->
      </div>
      <div
        class="multiselect__option"
      >
        <div
          class="checkbox-element multiselect__checkbox checkbox-element--normal checkbox-element--dark"
          data-id="option-2"
          tabindex="0"
          title="option 2"
        >
          <div
            class="checkbox-element__check-row"
            title="option 2"
          >
            <div
              class="checkbox-element__check-wrapper"
            >
              <div
                class="checkbox-element__square"
              />
               
              <div
                class="checkbox-element__check"
              />
            </div>
             
            <div
              class="checkbox-element__label-text"
            >
              
                    option 2
                
            </div>
          </div>
           
          <!---->
        </div>
         
        <!---->
      </div>
      <div
        class="multiselect__option"
      >
        <div
          class="checkbox-element multiselect__checkbox checkbox-element--normal checkbox-element--dark"
          data-id="option-3"
          tabindex="0"
          title="option 3"
        >
          <div
            class="checkbox-element__check-row"
            title="option 3"
          >
            <div
              class="checkbox-element__check-wrapper"
            >
              <div
                class="checkbox-element__square"
              />
               
              <div
                class="checkbox-element__check"
              />
            </div>
             
            <div
              class="checkbox-element__label-text"
            >
              
                    option 3
                
            </div>
          </div>
           
          <!---->
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="multiselect__options"
    style="visibility: hidden; position: absolute;"
  >
    <div
      class="multiselect__option"
    >
      <div
        class="checkbox-element multiselect__checkbox checkbox-element--normal checkbox-element--dark"
        data-id="option-1"
        tabindex="0"
        title="option 1"
      >
        <div
          class="checkbox-element__check-row"
          title="option 1"
        >
          <div
            class="checkbox-element__check-wrapper"
          >
            <div
              class="checkbox-element__square"
            />
             
            <div
              class="checkbox-element__check"
            />
          </div>
           
          <div
            class="checkbox-element__label-text"
          >
            
                option 1
            
          </div>
        </div>
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="multiselect__option"
    >
      <div
        class="checkbox-element multiselect__checkbox checkbox-element--normal checkbox-element--dark"
        data-id="option-2"
        tabindex="0"
        title="option 2"
      >
        <div
          class="checkbox-element__check-row"
          title="option 2"
        >
          <div
            class="checkbox-element__check-wrapper"
          >
            <div
              class="checkbox-element__square"
            />
             
            <div
              class="checkbox-element__check"
            />
          </div>
           
          <div
            class="checkbox-element__label-text"
          >
            
                option 2
            
          </div>
        </div>
         
        <!---->
      </div>
       
      <!---->
    </div>
    <div
      class="multiselect__option"
    >
      <div
        class="checkbox-element multiselect__checkbox checkbox-element--normal checkbox-element--dark"
        data-id="option-3"
        tabindex="0"
        title="option 3"
      >
        <div
          class="checkbox-element__check-row"
          title="option 3"
        >
          <div
            class="checkbox-element__check-wrapper"
          >
            <div
              class="checkbox-element__square"
            />
             
            <div
              class="checkbox-element__check"
            />
          </div>
           
          <div
            class="checkbox-element__label-text"
          >
            
                option 3
            
          </div>
        </div>
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;
